koku/api/report/ocp/query_handler.py

```diff
@@ -305,78 +325,78 @@ def execute_query(self):  # noqa: C901
+
+    def _generate_capacity_subsets(self, capacity_level, group_by):
+        """Calculate capacity over the timescope and resolution"""
+        _capacity = None
+        annotations = self._mapper.report_type_map.get("capacity_aggregate", {}).get(capacity_level)
+        if not annotations:
+            return None
         cap_key = list(annotations.keys())[0]
-        _capacity = CapacitySubsets(key=cap_key, level_key=level_key, resolution=self.resolution)
         q_table = self._mapper.query_table
         LOG.debug(f"Using query table: {q_table}")
         query = q_table.objects.filter(self.query_filter)
         if self.query_exclusions:
             query = query.exclude(self.query_exclusions)
         with tenant_context(self.tenant):
-            cap_data = query.values(*group_list).annotate(**annotations)
+            cap_data = query.values(*["usage_start", group_by]).annotate(**annotations)
             for entry in cap_data:
-                level_value = entry.get(level_key, "")
+                if not _capacity:
+                    _capacity = CapacitySubsets(
+                        key=cap_key, level_key=capacity_level, count_units=entry.get("capacity_count_units")
+                    )
+                level_value = entry.get(capacity_level, "")
                 usage_start = entry.get("usage_start", "")
+                if self.resolution == "daily" and isinstance(usage_start, datetime.date):
+                    usage_start = usage_start.isoformat()
+                if self.resolution == "monthly":
+                    usage_start = usage_start.strftime("%Y-%m")
                 cap_value = entry.get(cap_key, 0)
-                if cap_value is None:
-                    cap_value = 0
                 _capacity.add(cap_value, usage_start, level_value)
+                _capacity.add_instance_count(entry.get("capacity_count"), usage_start, level_value)
         return _capacity
 
-    def _get_node_capacity(self, query_data, _capacity):
-        """Calculate node capacity for all nodes over the date range."""
-        for row in query_data:
-            if row.get("node"):
-                _calculate_unused(row)
-            elif self.resolution == "daily":
-                row[_capacity.key] = _capacity.resolution_total.get(row.get("date"), Decimal(0))
-            elif self.resolution == "monthly" and not self.parameters.get("start_date"):
-                row[_capacity.key] = _capacity.total
-            else:
-                row_date = datetime.datetime.strptime(row.get("date"), "%Y-%m").month
-                row[_capacity.key] = _capacity.resolution_total.get(row_date, Decimal(0))
-        return query_data, {_capacity.key: _capacity.total}
-
     def _get_cluster_capacity(self, query_data, _capacity):
         """Calculate the cluster capacity."""
         for row in query_data:
-            cluster_list = row.get("clusters")
+            row_date = row.get("date")
+            # I am not convinced we need this:
```

Comment:
I am not convinced we need this comment :smiling_imp:

Reply:
💀 lol 🙇

Reply:
```suggestion
```